src/DotNetToolkit.Repository/RepositoryBaseAsync.cs

```diff
@@ -141,7 +150,7 @@ public Task<int> CountAsync(ISpecification<TEntity> criteria, CancellationToken
         /// <returns>The <see cref="System.Threading.Tasks.Task" /> that represents the asynchronous operation, containing the number of entities that satisfied the criteria specified by the <paramref name="predicate" /> in the repository..</returns>
         public Task<int> CountAsync(Expression<Func<TEntity, bool>> predicate, CancellationToken cancellationToken = new CancellationToken())
         {
-            return CountAsync(new Specification<TEntity>(predicate), cancellationToken);
+            return CountAsync(predicate == null ? null : new Specification<TEntity>(predicate), cancellationToken);
```

Comment:
There was an issue here where an exception would be thrown if the predicate is null